lib/yarp/parser_gem_compat.rb

```diff
@@ -0,0 +1,259 @@
+      end
+
+      def visit_if_node(node)
+        s(:if, convert(node.predicate), convert_or_nil(node.statements), convert_or_nil(node.consequent))
+      end
+
+      def visit_unless_node(node)
+        s(:if, convert(node.predicate), convert_or_nil(node.consequent), convert_or_nil(node.statements))
+      end
+
+      def visit_else_node(node)
+        convert(node.statements)
+      end
+
+      def visit_next_node(node)
+        s(:next)
+      end
+
+      def visit_module_node(node)
+        s(:module, convert(node.constant_path), convert_or_nil(node.body))
+      end
+
+      def visit_class_node(node)
+        s(:class, convert(node.constant_path), convert_or_nil(node.superclass), convert_or_nil(node.body))
+      end
+
+      def visit_def_node(node)
+        parameters = convert_or_nil(node.parameters) || s(:args)
+        if node.receiver
+          s(:defs, convert(node.receiver), node.name.to_sym, parameters, convert_or_nil(node.body))
+        else
+          s(:def, node.name.to_sym, parameters, convert_or_nil(node.body))
+        end
+      end
+
+      def visit_block_node(node)
+        parameters = convert_or_nil(node.parameters) || s(:args)
+        [parameters, convert_or_nil(node.body)]
+      end
+
+      def visit_block_parameters_node(node)
+        raise unless node.locals.empty?
+        convert(node.parameters)
+      end
+
+      def visit_parameters_node(node)
+        children = [*convert_array(node.requireds), *convert_array(node.optionals)]
+        children << convert(node.rest) if node.rest
+        children.concat convert_array(node.posts)
+        children.concat convert_array(node.keywords)
+        children << convert(node.keyword_rest) if node.keyword_rest
+        children << convert(node.block) if node.block
+        s(:args, *children)
+      end
+
+      def visit_required_parameter_node(node)
+        s(:arg, node.constant_id)
+      end
+
+      def visit_optional_parameter_node(node)
+        s(:optarg, node.constant_id, convert(node.value))
+      end
+
+      def visit_rest_parameter_node(node)
+        s(:restarg, node.name.to_sym)
+      end
+
+      def visit_block_parameter_node(node)
+        s(:blockarg, node.name.to_sym)
+      end
+
+      def visit_array_node(node)
+        s(:array, *convert_array(node.elements))
+      end
+
+      def visit_hash_node(node)
+        s(:hash, *convert_array(node.elements))
+      end
+
+      def visit_keyword_hash_node(node)
+        visit_hash_node(node)
+      end
+
+      def visit_assoc_node(node)
+        s(:pair, convert(node.key), convert(node.value))
+      end
+
+      def visit_assoc_splat_node(node)
+        s(:kwsplat, convert(node.value))
+      end
+
+      def visit_call_node(node)
+        arguments = node.arguments ? convert_array(node.arguments.arguments) : []
+        converted = s(:send, convert_or_nil(node.receiver), node.name.to_sym, *arguments)
+        if node.block
+          s(:block, converted, *convert(node.block))
```

Comment:
This is quite weird in the parser gem wrap the `:send` with a `:block` when passing a block, much nicer in YARP.
Ignoring that, the parser gem also has a single node for CallNode, nice.